### ISSUE TITLE
SSP for 800-171 appliance apps print out the wrong control_catalog[control.lower()]['family_id'] for 3.3.1

### ISSUE THREAD

**Issue by eborchert80** [GovReady/govready-q#1630]
**Describe the bug as a Story**
I noticed while looking at an SSP for "General IT System ATO for 800-171" application that the controls for 3.3.1 and 3.3.2 mapped to the wrong 800-171 families. For example,
```
3.1 - Access Control
3.3.1 - Create and retain system audit logs and records to the extent needed to enable the monitoring, analysis, investigation, and reporting of unlawful or unauthorized system activity.
Create and retain system audit logs and records to the extent needed to enable the monitoring, analysis, investigation, and reporting of unlawful or unauthorized system activity.

What is the solution and how is it implemented?
3.2 - Awareness and Training
3.3.2 - Ensure that the actions of individual system users can be uniquely traced to those users so they can be held accountable for their actions.
Ensure that the actions of individual system users can be uniquely traced to those users so they can be held accountable for their actions.
```
It turns out this bug is caused by an issue in get_group_id_by_control_id().

```
def get_group_id_by_control_id(self, control_id):
<snip>
         group_ids = self.get_group_ids()
         for group_id in group_ids:
            if group_id.lower() in control_id.lower(): <==
```
It turns out some control_id like '3.3.1' will eval to true when the group_id is '3.1' , for instance.

```
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> '3.1' in '3.3.1'
True
>>>
```
I wrote up a quick fix in my own environment that I am interested in getting feedback

**To Reproduce**
1.  Start a 800-171 project
2.  Click on "System Security Plan"
3. Scroll down to '3.3.1' controls
4. See it improperly mapped to the wrong control family
5.  Also occurs for  3.10.x

**Expected behavior**
3,3.1 should map to 3.3 - Audit and Accountability

**Screenshots**
Attaching a screenshot with the issue and with the fix
[issue]
![image](https://user-images.githubusercontent.com/61212480/122144015-d64af500-ce0f-11eb-9c9f-3d8d8897d359.png)
[fix]
![image](https://user-images.githubusercontent.com/61212480/122144053-e662d480-ce0f-11eb-94fb-898022384e07.png)

**Desktop (please complete the following information):**
 - Virtual Machine?
 - Screen size

## Developer hints

Look at controls/oscal.py , function get_group_id_by_control_id()

## Acceptance criteria 

800-171 should have the correct family mappings

**Additional context**
The fix I am using tmp in my env is the following code. It appears to work for both 800-53 and 800-171 controls.  Improvements could be made, however, looking for feedback

```
    def get_group_id_by_control_id(self, control_id):
        """Return group id given id of a control"""

        # For 800-53, 800-171, we can match by first few characters of control ID
        group_ids = self.get_group_ids()
        for group_id in group_ids:
            # 3.1 = 3.1
            if group_id.lower() == control_id.rsplit('.',1)[0].lower():
                return group_id
            ## ac == ac
            elif group_id.lower() == control_id.split('-')[0].lower():
                return group_id
```

**Comment by davidpofo** [comment]
@eborchert80 Thank you very much for bring this to our attention! I went ahead and just used `==` instead of `in`. Try it out yourself
 ```
>>> '3.1' == '3.3.1'
False
>>>
```
Here's the [pull request](https://github.com/GovReady/govready-q/pull/1632). Will get this merged to develop and then it will be in our next release!

**Comment by eborchert80** [comment]
Hi david,

I tested the fix for #1630 but I am running into an issue for both 800-53
and 800-171. See below screenshot
See attachment
[image: image.png]
This is why I restored to this other fix

        for group_id in group_ids:
            # 3.1 = 3.1
            if group_id.lower() == control_id.rsplit('.',1)[0].lower():
                return group_id
            ## ac == ac
            elif group_id.lower() == control_id.split('-')[0].lower():
                return group_id

But let me know. I may be missing something
